homeworks/rostyslav.havuka_RostyslavHavuka/11-dependency-injection/vehicle/src/app/pipes/protocol-car.pipe.ts

```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Car } from '../model';
+
+@Pipe({
+  name: 'protocolCar'
+})
+export class ProtocolCarPipe implements PipeTransform {
+  transform(car: Car): string {
+    return car.post.protocol
```

Comment:
Also an extra pipe. You don't actually need a pipe if it gets only one field and returns it.
The pipe's role is to combine complicated and time expensive (or not) logic inside.

In your case, you don't have complicated logic at all.

**Suggestion:**
I believe you can use this pipe to encode URL as mentioned in the task 😉